collections/without_all_test.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../testing/asserts.ts";
+import { withoutAll } from "./without_all.ts";
+
+function withoutAllTest<I>(
+  input: Array<I>,
+  excluded: Array<I>,
+  expected: Array<I>,
+  message?: string,
+) {
+  const actual = withoutAll(input, excluded);
+  assertEquals(actual, expected, message);
+}
+
```

Comment:
A "no mutation" test making sure that the original array that gets passed in does not get changed would be nice - we have that in all other function test suites as well

Reply:
The `filter()` method creates `a new array` with all elements that pass the test implemented by the provided function. so if we want `no mutation` we must create own filter function in order to return the same array instance.

Reply:
I meant to write a test that makes sure that the array you pass in does **not** get changed. This is true with the current implementation, I just say we should have a test to ensure that it stays so.

```suggestion
Deno.test({
  name: "[collections/withoutAll] no mutation",
  fn() {
    const array = [1, 2, 3, 4];
    withoutAll(array, [2, 3]);
    
    assertEquals(array, [1, 2, 3, 4]);
  },
});
```